crates/router/src/core/payments/helpers.rs

```diff
@@ -917,6 +917,66 @@ pub(crate) fn validate_payment_method_fields_present(
         },
     )?;
 
+    utils::when(
+        !matches!(req.payment_method, Some(api_enums::PaymentMethod::Card))
```

Comment:
I noticed this not but this `matches!()` will also return `false` for when `req.payment_method` is `None` which is basically asserting that even if `payment_method` is not given, `payment_method_type` should be given which is incorrect. The following should fix this afaik
```suggestion
        !matches!(req.payment_method, Some(api_enums::PaymentMethod::Card) | None)
```